packages/web-app-files/tests/unit/components/SideBar/FileInfo.spec.js

```diff
@@ -0,0 +1,184 @@
+import Vuex from 'vuex'
+import DesignSystem from 'owncloud-design-system'
+import stubs from '@/tests/unit/stubs/index.js'
+import { createLocalVue, mount } from '@vue/test-utils'
+import FileInfo from '@files/src/components/SideBar/FileInfo.vue'
+
+import GetTextPlugin from 'vue-gettext'
+
+const localVue = createLocalVue()
+localVue.use(DesignSystem)
+localVue.use(Vuex)
+localVue.use(GetTextPlugin, {
+  translations: 'does-not-matter.json',
+  silent: true
+})
+
+const filesPersonalRoute = { name: 'files-personal' }
+const filesPublicRoute = { name: 'files-public-list' }
+
+function getWrapper(
+  route,
+  { filename = 'testfile', extension, type = 'file', publicLink = false }
+) {
+  const mountStubs = { ...stubs, 'oc-button': false }
+  const mountOptions = {
+    localVue,
+    store: createStore({}, filename, extension, type),
+    stubs: mountStubs,
+    mocks: {
+      $route: route,
+      publicPage: () => publicLink
+    },
+    propsData: {
+      isContentDisplayed: true
+    }
+  }
+  return mount(FileInfo, mountOptions)
+}
+
+const selectors = {
+  fileNameItem: '.file_info h2',
+  favoritesButton: '.file_info button.file_info__favorite',
+  fileDescription: '.file_info__body div'
+}
+
+describe('SideBar', () => {
+  describe('when the user on files personal route', () => {
+    it('should show file name', () => {
+      const wrapper = getWrapper(filesPersonalRoute, {
+        filename: 'testfile',
+        extension: 'jpg'
+      })
+
+      const fileName = wrapper.find(selectors.fileNameItem)
+      expect(fileName.isVisible()).toBeTruthy()
+      expect(fileName.text()).toBe('testfile.jpg')
+    })
+
+    it('should show favorites button in files personal page', () => {
+      const wrapper = getWrapper(filesPersonalRoute, {
+        filename: 'testfile',
+        extension: 'jpg'
+      })
+
+      const favoritesButton = wrapper.find(selectors.favoritesButton)
+      expect(favoritesButton.exists()).toBeTruthy()
+    })
+
+    it('should not show favorites button in public link page', () => {
+      const wrapper = getWrapper(filesPublicRoute, {
+        filename: 'testfile',
+        extension: 'jpg',
+        publicLink: true
+      })
+
+      const favoritesButton = wrapper.find(selectors.favoritesButton)
+      expect(favoritesButton.exists()).toBeFalsy()
+    })
+
+    it('should allow user to favorite a resource', async () => {
```

Comment:
IMO this should be an integration test.

Reply:
just for reference that is the same discussion as here https://github.com/owncloud/web/pull/5557/files#r693829700